### ISSUE TITLE
Wire.requestFrom() should not be followed by Wire.endTransmission().

### ISSUE THREAD

**Issue by Koepel** [chrisxkeith/ir-qwiic-example#1]
In the file "ir-qwiic-example/src/ir-qwiic-example.ino", there is a Wire.endTransmission() after a Wire.requestFrom(). You may remove that Wire.endTransmission(). It is at the moment at line 1018.
Explanation: [Common-mistakes#2](https://github.com/Koepel/How-to-use-the-Arduino-Wire-library/wiki/Common-mistakes#2)
See also my [alternative explanation of the functions of the Wire library](https://github.com/Koepel/How-to-use-the-Arduino-Wire-library/wiki/Explanation-of-the-functions-of-the-Wire-library).

What is the parameter 'len' for in the getRegister() function ? That functions works only well if 'len' is 2.

**Comment by chrisxkeith** [comment]
Thank you for seeing this. This is code that I got from https://learn.sparkfun.com/tutorials/qwiic-grid-eye-infrared-array-amg88xx-hookup-guide/all. I have to plead guilty to the fact that I didn't look at it (and thus don't really understand it). I will remove line 1018 as you suggested.

>> What is the parameter 'len' for in the getRegister() function ? That functions works only well if 'len' is 2.

If all the calls to that function should be changed, I will do that. Otherwise, if you could tell me which specific calls to change, I will change only those calls.

Best regards,
Chris

**Comment by Koepel** [comment]
Looking at `getRegister()`, it's just wrong. So I went over to Sparkfun to find out that I already wrote an issue: https://github.com/sparkfun/SparkFun_GridEYE_Arduino_Library/issues/3

I prefer `getRegister16()` and `getRegister8()`. A `len` parameter makes sense if an array of data is requested from a sensor.

A quick fix could be:
```
// The parameter "len" may only be 1 (to retrieve a byte) or 2 (for a signed 16-bit integer).
int16_t GridEYE::getRegister(unsigned char reg, int8_t len)
{
  int16_t result = 0;     // default 0, if the sensor was not connected

  _i2cPort->beginTransmission(_deviceAddress);
  _i2cPort->write(reg);
  _i2cPort->endTransmission(false);    // 'false' for a repeated start

  _i2cPort->requestFrom((uint8_t)_deviceAddress, (uint8_t)len);
  if (_i2cPort->available() == len)    // got valid data ?
  {
    if (len == 1)
    {
      result = (int16_t) _i2cPort->read();
    }
    else if (len == 2)
    {
      int lsb = _i2cPort->read();
      int msb = _i2cPort->read();

      // concat bytes
      result = (int16_t) (msb << 8 | lsb);
    }
  }
  return result;
}
```

This will fix the extra transaction on the I2C bus and it will fix reading data that is not requested, but it is fixing a wrong function with code that makes me unhappy. The `len` can only be 1 or 2.

The Wire.read() is part of the Stream class and returns an 'int' on every platform, so that is at least 16 bits. Shifting to the left is the same for signed and unsigned, so the concatenation should work.

In my opinion, the function `isFramerate10FPS()` does not work, and with the function above it suddenly will work, so the library will change its behaviour.

**Comment by chrisxkeith** [comment]
Fixed by commit 57b5ca2881ab5253983b7d6346631faa21ddbe6b . Thanks!
